saml/src/main/java/com/linecorp/armeria/server/saml/SamlAssertionConsumerConfigBuilder.java

```diff
@@ -61,6 +71,7 @@ public SamlServiceProviderBuilder and() {
      * Builds a {@link SamlAssertionConsumerConfig}.
      */
     SamlAssertionConsumerConfig build() {
+        checkState(endpoint != null, "The endpoint must not be null.");
```

Comment:
Q) should we prefer `requireNonNull` generally when checking nullability?
i.e. `NullPointerException` vs. `IllegalStateException`

Reply:
Good question. `NullPointerException` looks good to me.

Reply:
I think either is fine in the `build` method. 😉 
These are some of the examples that use `checkState`.
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java#L468
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/auth/OAuth1aTokenBuilder.java#L220
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/AbstractHttpRequestBuilder.java#L431

Reply:
If it is stateful, some explanation with `checkState` also makes sense to me.